### ISSUE TITLE
NullPointerException on Matchers.eq(null)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [dlsgusrn7577/mockito#154]
```
With the follogin line in a test we have the following :

connectionListener.onInit(anyString(), anyString(), anyString(), 
any(Instant.class), eq((String) null))).thenReturn(initiateEvent);


Up to mockito 1.8.0 the test passed.
With Mockito 1.8.1 I got a NullPointerException on Matchers.eq

This bug on this minor revision of mockito forces us to rewrite many 
passing tests.

By the way NPE should be avoided on this API or at least the mockito 
framework should report this as a misusage (as we fixed this using 
isNull()).
```

Original issue reported on code.google.com by `brice.du...@gmail.com` on 3 Dec 2009 at 1:50

**Comment by GoogleCodeExporter** [comment]
```
It's a bug - we'll fix it.

>This bug on this minor revision of mockito forces us to rewrite many 
>passing tests.

Next time to avoid this situation you can run your tests against mockito release
candidate. This way you can protect yourself from breaking existing tests + you 
can
help us releasing the target release without bugs :)
```

Original comment by `szcze...@gmail.com` on 3 Dec 2009 at 2:31

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision r1786.
```

Original comment by `szcze...@gmail.com` on 3 Dec 2009 at 9:15
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision r1788.
```

Original comment by `szcze...@gmail.com` on 3 Dec 2009 at 11:05

**Comment by GoogleCodeExporter** [comment]
```
Thanks a lot
```

Original comment by `brice.du...@gmail.com` on 4 Dec 2009 at 8:39

**Comment by GoogleCodeExporter** [comment]
Original comment by `szcze...@gmail.com` on 12 Dec 2009 at 2:20
- Added labels: **Milestone-Release1.8.2**

**Comment by GoogleCodeExporter** [comment]
```
Released in 1.8.2 on 12-12-2009
```

Original comment by `szcze...@gmail.com` on 13 Dec 2009 at 10:41

**Comment by GoogleCodeExporter** [comment]
Original comment by `brice.du...@gmail.com` on 2 Dec 2012 at 10:21
- Added labels: **Matchers**
